articles/how-to-use-local-storage-with-JavaScript/index.md

```diff
@@ -0,0 +1,248 @@
+
+removeButton invokes `removeItem()`.
+
+clearButton calls the `clearStorage()`.
+
+Here is the `main.js` file with the required functions.
+
+```javascript
+function store(){ //stores items in the local storage
+        var brand = document.getElementById('carBrand').value;
+        var price = document.getElementById('carPrice').value;
+    
+        const car = {
+            brand: brand,
+            price: price,
+        }
+        
+        window.localStorage.setItem('car',JSON.stringify(car));  
+        //converting object to string
+}
+
+function retrieveRecords(){ //retrieves items in the localStorage
+    console.log("retrive records");
+    var records = window.localStorage.getItem('car');
+    var paragraph = document.createElement("p");
+    var infor = document.createTextNode(records);
+    paragraph.appendChild(infor);
+    var element = document.getElementById("retrieve");
+    element.appendChild(paragraph);
+}
+
+
+function removeItem(){//deletes item from localStorage
+    localStorage.removeItem('car')
+    console.log("remove items");
+}
+
+function clearStorage(){ //clears the entire localStorage
+    localStorage.clear()
+    console.log("clear records");
+}
+
+window.onload =function(){ //ensues the page is loaded before functions are executed.
+    document.getElementById("carForm").onsubmit = store
+
+    document.getElementById("clearButton").onclick = clearStorage
+
+    document.getElementById("removeButton").onclick = removeItem
+
+    document.getElementById("retrieveButton").onclick = retrieveRecords
+
+}
+
+```
+
+As shown above, the functions will only be clickable after the page has finished loading. This is specified by the `window.onload` method.
+
+Ensure that the `main.js` file is referenced in the `index.html` file by pasting the statement below in the `head` section.
+
+```
+  <script type="text/javascript" src="main.js"></script>
+```
+
+### 3. Results
+The following video shows how the site looks, as well as the invocation of localStorage methods.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/nzitoIbsT2g" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+### When to use local storage
+You should only use localStorage when storing insensitive information. This is because third-party individuals can easily access the information. localStorage can help in storing temporary data before it is pushed to the server. It is important to clear localStorage once this operation is completed.
+
+### Limitations
+The major limitations of localStorage are:
+
+- Insecure data
+
+- Synchronous operations
+
+- Limited storage capacity
+
+#### Conclusion
+You are now familiar with the different functionalities of localStorage. In case, you have not understood any concept, feel free to go through the localStorage functions again.
```

Comment:
`In case, you have not understood any concept,`

Replace have not with didn't

Reply:
Revised